Projects/Server/Items/Container.Enumerable.cs

```diff
@@ -0,0 +1,226 @@
+using Server.Collections;
+using Server.Common.Filters;
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Server.Items;
+
+public partial class Container
+{
+    private static readonly Queue<Container> containersForGetItem = new();
```

Comment:
We can use `PooledRefQueue` instead.

Reply:
Yep, this static queue is used only for "FirstOrDefault": I tried with a static queue because in that case we don't need a temporary queue that survive outside the single method execution and maybe a standard queue (in this single thread code execution) could be better than ask from the pool a queue for every execution.
But I can switch to the pooledrefqueue: in this case the implementation could be easier because it's like "call the enumerator and do MoveNext one time".